src/libraries/System.Reflection.DispatchProxy/src/System/Reflection/DispatchProxyGenerator.cs

```diff
@@ -823,54 +823,12 @@ private sealed class MethodInfoEqualityComparer : EqualityComparer<MethodInfo>
 
                 public sealed override bool Equals(MethodInfo? left, MethodInfo? right)
                 {
-                    if (ReferenceEquals(left, right))
-                        return true;
-
                     if (left == null)
                         return right == null;
                     else if (right == null)
                         return false;
 
-                    // TODO: switch to use MemberInfo.MetadataToken here.
-                    // It compares honestly referring ECMA-335 I.8.6.1.6 Signature Matching.
-                    if (!Equals(left.DeclaringType, right.DeclaringType))
-                        return false;
-
-                    if (!Equals(left.ReturnType, right.ReturnType))
-                        return false;
-
-                    if (left.CallingConvention != right.CallingConvention)
-                        return false;
-
-                    if (left.IsStatic != right.IsStatic)
-                        return false;
-
-                    if (left.Name != right.Name)
-                        return false;
-
-                    Type[] leftGenericParameters = left.GetGenericArguments();
-                    Type[] rightGenericParameters = right.GetGenericArguments();
-                    if (leftGenericParameters.Length != rightGenericParameters.Length)
-                        return false;
-
-                    for (int i = 0; i < leftGenericParameters.Length; i++)
-                    {
-                        if (!Equals(leftGenericParameters[i], rightGenericParameters[i]))
-                            return false;
-                    }
-
-                    ParameterInfo[] leftParameters = left.GetParameters();
-                    ParameterInfo[] rightParameters = right.GetParameters();
-                    if (leftParameters.Length != rightParameters.Length)
-                        return false;
-
-                    for (int i = 0; i < leftParameters.Length; i++)
-                    {
-                        if (!Equals(leftParameters[i].ParameterType, rightParameters[i].ParameterType))
-                            return false;
-                    }
-
-                    return true;
+                    return left.Module.Name == right.Module.Name && left.MetadataToken == right.MetadataToken;
```

Comment:
The modules can have same name, but different identity. This condition would not work correctly in that case.

Should the whole MethodInfoEqualityComparer type be deleted and the dictionary switched to use the default comparer? It is not clean why this implementation is trying to achieve that the default EqualityComparer would not.

Reply:
The default comparator will compare the reference of the object. It cannot be judged correctly when used here, so need to use `MethodInfoEqualityComparer`

Reply:
The default EqualityComparer will call Equals method to compare the types: https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/EqualityComparer.cs#L135

MethodInfo.Equals should do exactly what the existing code is trying to achieve.

Reply:
I did not find information about `MethodInfo.Equals` override from `Object.Equals`